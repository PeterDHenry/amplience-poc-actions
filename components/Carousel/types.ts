import { HeroProps } from '../Hero/types'

export type CarouselProps = {
  heroBannerList: HeroProps[]
}

export const defaultProps: CarouselProps = {
  heroBannerList: [
    {
      _meta: {
        name: 'Default ride like a pro banner',
        schema: 'https://wiggle.com/hero-banner',
        deliveryId: '18a7eee8-6c7a-4a92-8a76-29eb936679a8',
      },
      background: {
        _meta: {
          schema:
            'http://bigcontent.io/cms/schema/v1/core#/definitions/image-link',
        },
        id: 'ef717f48-99e2-4717-89da-6f906056eae1',
        name: 'CCC-11-5-a',
        endpoint: 'wigglecrcnonprd',
        defaultHost: 'cdn.media.amplience.net',
      },
      textAndCTA: {
        headline: 'Default ride like a pro',
        callToActionURL: 'https://www.wiggle.co.uk/pro-riding-for-summer',
        callToActionText: 'Default shop now',
      },
    },
    {
      _meta: {
        name: 'Default Tailored for women banner',
        schema: 'https://wiggle.com/hero-banner',
        deliveryId: 'ecba6af1-17ce-4799-9d20-146b307af92d',
      },
      background: {
        _meta: {
          schema:
            'http://bigcontent.io/cms/schema/v1/core#/definitions/image-link',
        },
        id: '9c6bd46b-21e2-4720-8be6-3fb4b9be6079',
        name: 'homepageh1d',
        endpoint: 'wigglecrcnonprd',
        defaultHost: 'cdn.media.amplience.net',
      },
      textAndCTA: {
        headline: 'Default Tailored for women',
        strapline: 'Your best-ever sports collection',
        callToActionText: 'Default Shop now',
        callToActionURL: 'https://www.wiggle.co.uk/womens-collection',
      },
    },
    {
      _meta: {
        name: 'Default Discover different banner',
        schema: 'https://wiggle.com/hero-banner',
        deliveryId: 'a6cac415-3411-4481-a741-c7d4e1edbff2',
      },
      background: {
        _meta: {
          schema:
            'http://bigcontent.io/cms/schema/v1/core#/definitions/image-link',
        },
        id: 'a11cb660-fd4a-498f-8222-022edfeb479b',
        name: 'homepage-h-d',
        endpoint: 'wigglecrcnonprd',
        defaultHost: 'cdn.media.amplience.net',
      },
      textAndCTA: {
        headline: 'Default Discover different',
        callToActionText: 'Default Find out more',
        callToActionURL: 'https://www.wiggle.co.uk/discover-different',
      },
    },
  ],
}
