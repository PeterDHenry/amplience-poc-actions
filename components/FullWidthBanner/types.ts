export type FullWidthBannerProps = {
  _meta: {
    name: string
    schema: string
    deliveryId: string
  }
  image: { name: string }
  textAndCTA: {
    callToActionURL: string
    callToActionText: string
    headline: string
    strapline: string
  }
}

export const defaultProps: FullWidthBannerProps = {
  _meta: {
    name: 'string',
    schema: 'string',
    deliveryId: '1543512973',
  },
  image: { name: 'homepageh1d' },
  textAndCTA: {
    callToActionURL: 'https://www.wiggle.co.uk/run-new-season-2021',
    callToActionText: 'Default cta text',
    headline: 'Default header',
    strapline: 'Default subheader',
  },
}
