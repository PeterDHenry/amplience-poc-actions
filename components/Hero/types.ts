export type HeroProps = {
  _meta: {
    name: string
    schema: string
    deliveryId: string
  }
  cssClass?: string
  background: {
    _meta: {
      schema: string
    }
    id: string
    name: string
    endpoint: string
    defaultHost: string
  }
  //color: string
  //contrastColor: string
  textAndCTA: {
    callToActionURL: string
    callToActionText: string
    headline: string
    strapline?: string
  }
}

export const defaultProps: HeroProps = {
  _meta: {
    name: 'Ride like a pro banner',
    schema: 'https://wiggle.com/hero-banner',
    deliveryId: '18a7eee8-6c7a-4a92-8a76-29eb936679a8',
  },
  background: {
    _meta: {
      schema: 'http://bigcontent.io/cms/schema/v1/core#/definitions/image-link',
    },
    id: 'ef717f48-99e2-4717-89da-6f906056eae1',
    name: 'CCC-11-5-a',
    endpoint: 'wigglecrcnonprd',
    defaultHost: 'cdn.media.amplience.net',
  },
  textAndCTA: {
    headline: 'Ride like a pro',
    callToActionURL: 'https://www.wiggle.co.uk/pro-riding-for-summer',
    callToActionText: 'Shop now',
  },
}
