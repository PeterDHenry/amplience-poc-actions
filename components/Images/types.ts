export type LazyLoadImageProps = {
  altText: string
  animationDuration: number
  applyAspectRatio: boolean
  height: number
  id?: string
  width: number
}

export type HeroImageProps = {
  altText: string
  id: string
}

export const defaultLazyLoadImageProps: LazyLoadImageProps = {
  altText: 'Default alt text',
  id: 'homepage-h-d',
  animationDuration: 0.1,
  width: 1920,
  height: 650,
  applyAspectRatio: true,
}

export const defaultHeroImageProps: HeroImageProps = {
  altText: 'Default hero alt text',
  id: 'homepage-h-d',
}
