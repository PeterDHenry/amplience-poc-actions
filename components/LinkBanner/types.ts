export type LinkBannerProps = {
  textAndCTA: {
    callToActionURL: string
    callToActionText: string
    headline: string
    strapline: string
  }
  backgroundColour: string
  fontColour: string
}

export const defaultLinkBannerProps: LinkBannerProps = {
  textAndCTA: {
    callToActionURL: 'https://www.wiggle.co.uk/sc/wiggle-plus',
    callToActionText: 'Find out more',
    headline:
      'Wiggle+ unlimited next day delivery & FREE RETURNS for a year - only Â£9.99',
    strapline: '',
  },
  backgroundColour: '#F7F7F7',
  fontColour: '#3C3C3B',
}
