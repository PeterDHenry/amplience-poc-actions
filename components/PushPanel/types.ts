export interface PushPanelProps {
  height: number
  width: number
  _meta: {
    name: string
    schema: string
    deliveryId: string
  }
  mainImg: { name: string }
  color: string
  fontWeight: number
  textAndCTA: {
    callToActionURL: string
    callToActionText: string
    headline: string
    strapline: string
  }
}

export interface PushPanelListProps {
  pushPanelList: PushPanelProps[]
  stackMobileLayout?: boolean
}
