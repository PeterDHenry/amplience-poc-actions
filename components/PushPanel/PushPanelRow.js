import React from 'react';
import PushPanel from './PushPanel'

export default function CarouselsAll ({ pushPanelList }) {
    debugger
    return ( 
      <div className="container py-5">
        <div className="row">
          {( pushPanelList.map((panel)=>(
            <div className="col-xs-6 col-sm-4 w-pr-4 w-pl-2 w-pl-sm-4">
              <PushPanel  key={panel._meta.deliveryId}
                          {...panel}/>
            </div>
          )))}   
        </div>
      </div>
    )
};
  
