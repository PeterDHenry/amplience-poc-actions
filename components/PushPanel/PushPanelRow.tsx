import React from 'react'
import PushPanel from './PushPanel'
import { PushPanelProps, PushPanelListProps } from './types'

export default function PushPanelRow({
  pushPanelList,
  stackMobileLayout,
}: PushPanelListProps) {
  let cssClass = `w-pr-4 w-pl-2 w-pl-sm-4 stack-${stackMobileLayout}`
  switch (pushPanelList.length) {
    case 2: {
      cssClass = 'col-xs-12 col-sm-6 ' + cssClass
      break
    }
    case 3: {
      cssClass = 'col-xs-6 col-sm-4 ' + cssClass
      break
    }
    //TODO
    //default: {

    //break
    //}
  }
  return (
    <div className="container py-5">
      <div className="row">
        {pushPanelList.map((panel: PushPanelProps) => (
          <div key={panel._meta.deliveryId} className={cssClass}>
            <PushPanel {...panel} />
          </div>
        ))}
      </div>
    </div>
  )
}
