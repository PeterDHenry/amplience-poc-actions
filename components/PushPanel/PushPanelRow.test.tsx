//TODO best way to test with different counts hint this isn't it
import { render } from '@testing-library/react'
import PushPanelRow from './PushPanelRow'
import { PushPanelListProps } from './types'

describe('ThreePushPanel component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <PushPanelRow {...twoPushPanelListProps} />,
      {}
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it("three doesn't matches snapshot", () => {
    const { asFragment } = render(
      <PushPanelRow {...threePushPanelListProps} />,
      {}
    )
    expect(!asFragment()).toMatchSnapshot()
  })
})

const twoPushPanelListProps: PushPanelListProps = {
  pushPanelList: [
    {
      _meta: {
        name: 'name',
        schema: 'schema',
        deliveryId: '328756873',
      },
      mainImg: {
        name: 'homepageb3d2',
      },
      textAndCTA: {
        callToActionURL: 'https://www.wiggle.co.uk/discover-different',
        headline: 'Default headline',
        strapline: 'Default Strapline',
        callToActionText: 'Default call to action',
      },
      color: 'black',
      fontWeight: 100,
    },
    {
      _meta: {
        name: 'name',
        schema: 'schema',
        deliveryId: '328776873',
      },
      mainImg: {
        name: 'homepageb3d2',
      },
      textAndCTA: {
        callToActionURL: 'https://www.wiggle.co.uk/discover-different',
        headline: 'Default headline',
        strapline: 'Default Strapline',
        callToActionText: 'Default call to action',
      },
      color: 'black',
      fontWeight: 100,
    },
  ],
}

const threePushPanelListProps: PushPanelListProps = {
  pushPanelList: [
    {
      _meta: {
        name: 'name',
        schema: 'schema',
        deliveryId: '328756873',
      },
      mainImg: {
        name: 'homepageb3d2',
      },
      textAndCTA: {
        callToActionURL: 'https://www.wiggle.co.uk/discover-different',
        headline: 'Default headline',
        strapline: 'Default Strapline',
        callToActionText: 'Default call to action',
      },
      color: 'black',
      fontWeight: 100,
    },
    {
      _meta: {
        name: 'name',
        schema: 'schema',
        deliveryId: '328776873',
      },
      mainImg: {
        name: 'homepageb3d2',
      },
      textAndCTA: {
        callToActionURL: 'https://www.wiggle.co.uk/discover-different',
        headline: 'Default headline',
        strapline: 'Default Strapline',
        callToActionText: 'Default call to action',
      },
      color: 'black',
      fontWeight: 100,
    },
    {
      _meta: {
        name: 'name',
        schema: 'schema',
        deliveryId: '328776874',
      },
      mainImg: {
        name: 'homepageb3d2',
      },
      textAndCTA: {
        callToActionURL: 'https://www.wiggle.co.uk/discover-different',
        headline: 'Default headline',
        strapline: 'Default Strapline',
        callToActionText: 'Default call to action',
      },
      color: 'black',
      fontWeight: 100,
    },
  ],
}
