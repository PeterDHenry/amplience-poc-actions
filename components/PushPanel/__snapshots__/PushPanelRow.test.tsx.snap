// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThreePushPanel component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container py-5"
  >
    <div
      class="row"
    >
      <div
        class="col-xs-12 col-sm-6 w-pr-4 w-pl-2 w-pl-sm-4 stack-undefined"
      >
        <div
          class="PushPanel__Panel-sc-1magf3k-0 kDGwnJ"
        >
          <a
            class="PushPanel__PanelLink-sc-1magf3k-1 nQoaE"
            href="https://www.wiggle.co.uk/discover-different"
          >
            <img
              class="PushPanelImage__PanelImg-sc-14ujkc5-0 dpUuSb"
              src="https://cdn.media.amplience.net/i/wigglecrcnonprd/homepageb3d2"
            />
            <div
              class="PushPanel__PanelText-sc-1magf3k-2 kcYvrH"
            >
              <h2
                style="color: black; font-weight: 100;"
              >
                Default headline
              </h2>
              <h3
                style="color: black;"
              >
                Default Strapline
              </h3>
              <button
                style="color: black;"
              >
                Default call to action
              </button>
            </div>
          </a>
        </div>
      </div>
      <div
        class="col-xs-12 col-sm-6 w-pr-4 w-pl-2 w-pl-sm-4 stack-undefined"
      >
        <div
          class="PushPanel__Panel-sc-1magf3k-0 kDGwnJ"
        >
          <a
            class="PushPanel__PanelLink-sc-1magf3k-1 nQoaE"
            href="https://www.wiggle.co.uk/discover-different"
          >
            <img
              class="PushPanelImage__PanelImg-sc-14ujkc5-0 dpUuSb"
              src="https://cdn.media.amplience.net/i/wigglecrcnonprd/homepageb3d2"
            />
            <div
              class="PushPanel__PanelText-sc-1magf3k-2 kcYvrH"
            >
              <h2
                style="color: black; font-weight: 100;"
              >
                Default headline
              </h2>
              <h3
                style="color: black;"
              >
                Default Strapline
              </h3>
              <button
                style="color: black;"
              >
                Default call to action
              </button>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ThreePushPanel component three doesn't matches snapshot 1`] = `false`;
